cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -140,3 +144,62 @@ func WriteStaticPodToDisk(componentName, manifestDir string, pod v1.Pod) error {
 
 	return nil
 }
+
+// GetProbeAddress returns an IP address or 127.0.0.1 to use for liveness probes
+// in static pod manifests. For etcd, GetProbeAddress will use the first
+// listen-client-url if multiple urls are provided. If the listen-client-url
+// contains a hostname instead of an IP address, GetProbeAddress will use the
+// local resolver for name resolution or 127.0.0.1 if the name can not be resolved.
+func GetProbeAddress(cfg *kubeadmapi.MasterConfiguration, component string) string {
+	switch {
+	case component == kubeadmconstants.KubeAPIServer:
+		if cfg.API.AdvertiseAddress != "" {
+			return cfg.API.AdvertiseAddress
+		}
+	case component == kubeadmconstants.KubeScheduler:
+		if cfg.SchedulerExtraArgs != nil {
+			addr, exists := cfg.SchedulerExtraArgs["address"]
+			if exists {
+				return addr
+			}
+		}
+	case component == kubeadmconstants.KubeControllerManager:
+		if cfg.ControllerManagerExtraArgs != nil {
+			addr, exists := cfg.ControllerManagerExtraArgs["address"]
```

Comment:
same here

Reply:
Then the probe uses 0.0.0.0 and the probes pass.